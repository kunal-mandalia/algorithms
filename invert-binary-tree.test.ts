import invertBinaryTree from './invert-binary-tree'

describe('invert binary tree', () => {
  it('should reverse the nodes in a tree with height 2', () => {
    // arrange
    const inputBinaryTree = {
      value: 1,
      left: {
        value: 2,
        left: null,
        right: null
      },
      right: {
        value: 3,
        left: null,
        right: null
      }
    }

    // act
    // assert
    expect(invertBinaryTree(inputBinaryTree)).toMatchObject({
      value: 1,
      left: {
        value: 3,
        left: null,
        right: null
      },
      right: {
        value: 2,
        left: null,
        right: null
      }
    })
  })

  it('should reverse nodes in a tree with height 3', () => {
    // arrange
    /**
     * input:
     * 
     *     1
     *  2     3
     * 4,5   6, _
     * 
     * expected output:
     * 
     *     1
     *   3    2
     *  _,6  5, 4
     * 
     */
    const inputBinaryTree = {
      value: 1,
      left: {
        value: 2,
        left: {
          value: 4,
          left: null,
          right: null
        },
        right: {
          value: 5,
          left: null,
          right: null
        }
      },
      right: {
        value: 3,
        left: {
          value: 6,
          left: null,
          right: null
        },
        right: null
      }
    }
    // act
    // assert
    expect(invertBinaryTree(inputBinaryTree)).toMatchSnapshot()
  })
})