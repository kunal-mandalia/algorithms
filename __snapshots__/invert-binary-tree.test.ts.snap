// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`invert binary tree should reverse nodes in a tree with height 3 1`] = `
Object {
  "left": Object {
    "left": null,
    "right": Object {
      "left": null,
      "right": null,
      "value": 6,
    },
    "value": 3,
  },
  "right": Object {
    "left": Object {
      "left": null,
      "right": null,
      "value": 5,
    },
    "right": Object {
      "left": null,
      "right": null,
      "value": 4,
    },
    "value": 2,
  },
  "value": 1,
}
`;
